plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -937,11 +938,18 @@ def create_vm(module, proxmox, vmid, newid, node, name, memory, cpu, cores, sock
             kwargs['searchdomain'] = ' '.join(searchdomains)
 
     # VM tags are expected to be valid and presented as a comma/semi-colon delimited string
-    if module.params['tags']:
-        for tag in module.params['tags']:
-            if not re.match(r'^[a-z0-9_][a-z0-9_\-\+\.]*$', tag):
-                module.fail_json(msg='%s is not a valid tag' % tag)
-        kwargs['tags'] = ",".join(module.params['tags'])
+    if 'tags' in kwargs:
+        # Set tags to null if empty list
+        if not kwargs['tags']:
```

Comment:
Why have this as a special case, when this is already handled by the final `else:` branch?

Reply:
Yep, I got ahead of myself.
All that really needed to change was to pass the parameter as a keyword argument to the create_vm function.
That and testing if the 'tags' key is in kwargs is enough to have the same functionality with the benefits of version checking and sanitization already in the module. (I overlooked the following which made me think a default case was necessary.)
```python
879     # Sanitize kwargs. Remove not defined args and ensure True and False converted to int.
880     kwargs = dict((k, v) for k, v in kwargs.items() if v is not None)
```
Appreciate your insight and have an updated commit ready as soon as my tests are complete.

Reply:
Looks good now!